[tf.data] graduate choose_from_datasets API from experimental to tf.data.Dataset

kvignesh1420 opened the pull request:
This PR graduates the `tf.data.experimental.choose_from_datasets` API into `tf.data.Dataset.choose_from_datasets` by making the following changes:

- [x] Adds the deprecation decorator for the experimental API.
- [x] Add the `choose_from_datasets()` method to DatasetV2 class.
- [x]  Updates example in documentation with new API.
- [x]  Regenerate golden API's.
- [x]  Updated the sample_from_datasets usage in tests.
- [x]  Updated the RELEASE.md file

cc: @aaudiber

kvignesh1420 committed:
[tf.data] graduate choose_from_datasets API from experimental to tf.data.Dataset

kvignesh1420 committed:
remove unused imports from tests

kvignesh1420 committed:
regenerate golden API's

aaudiber commented on tensorflow/python/data/ops/dataset_ops.py:
```diff
@@ -3229,6 +3229,61 @@ def select_dataset_varying_logits(logits, seed):
+        to set it to `True`. Otherwise, the selected elements start off as
+        the user intends, but may change as input datasets become empty.
+        This can be difficult to detect since the dataset starts off looking
+        correct. Default to `False` for backward compatibility.
```
Can we make the default `True` now? The backward compatibility requirement no longer applies since we are migrating the API. We need to call out the change in default in the release notes and deprecation message, and tell users to set `stop_on_empty_dataset=False` to get the same as the experimental API.

We should do the same for `sample_from_datasets`, in a separate CL.

kvignesh1420 replied:
@aaudiber  I agree. I have made the necessary changes. Please take a look.

aaudiber reviewed (commented):
Thanks @kvignesh1420

kvignesh1420 committed:
default stop_on_empty_dataset to True

add tf.data breaking changes to RELEASE.md

fix api compatibility test failure

fix api compatibility test failures

kvignesh1420 commented:
@aaudiber can you please approve the pull request again?
